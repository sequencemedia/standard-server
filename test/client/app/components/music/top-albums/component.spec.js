import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import Component from '@sequencemedia/app/components/music/top-albums/component'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('@sequencemedia/app/components/common/order', () => {
  const MockOrder = () => (
    <div className='mock-order'>
      Mock Order
    </div>
  )

  return {
    __esModule: true,
    default: MockOrder
  }
})
jest.mock('@sequencemedia/app/components/common/reset', () => {
  const MockReset = () => (
    <div className='mock-reset'>
      Mock Reset
    </div>
  )

  return {
    __esModule: true,
    default: MockReset
  }
})
jest.mock('@sequencemedia/app/components/common/latest', () => {
  const MockLatest = () => (
    <div className='mock-latest'>
      Mock Latest
    </div>
  )

  return {
    __esModule: true,
    default: MockLatest
  }
})
jest.mock('@sequencemedia/app/components/common/albums', () => {
  const MockAlbums = () => (
    <div className='mock-albums'>
      Mock Albums
    </div>
  )

  return {
    __esModule: true,
    default: MockAlbums
  }
})

const MOCK_ITEMS = [
  {
    kind: 'MOCK KIND',
    url: 'MOCK URL',
    name: 'MOCK NAME',
    artworkUrl100: 'MOCK ARTWORK URL 100',
    artistUrl: 'MOCK ARTIST URL',
    artistName: 'MOCK ARTIST NAME',
    releaseDate: '2020-11-16',
    copyright: 'MOCK COPYRIGHT'
  }
]

describe('shinkansen-cogs/cogs/checkbox', () => {
  describe('<Component />', () => {
    describe('With required props', () => {
      const component = (
        <Component
          feedType='MOCK FEED TYPE'
          onMount={jest.fn()}
          onMountOrderBy={jest.fn()}
          onClickLatest={jest.fn()}
          onClickLatestOrderBy={jest.fn()}
        />
      )

      it('renders', () => {
        return expect(toJSON(shallow(component)))
          .toMatchSnapshot()
      })
    })

    describe('With additional props', () => {
      const component = (
        <Component
          feedType='MOCK FEED TYPE'
          order='MOCK ORDER'
          items={MOCK_ITEMS}
          onMount={jest.fn()}
          onMountOrderBy={jest.fn()}
          onClickLatest={jest.fn()}
          onClickLatestOrderBy={jest.fn()}
        />
      )

      it('renders', () => {
        return expect(toJSON(shallow(component)))
          .toMatchSnapshot()
      })
    })
  })
})
