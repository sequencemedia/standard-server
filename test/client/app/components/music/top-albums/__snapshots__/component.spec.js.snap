// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shinkansen-cogs/cogs/checkbox <Component /> With additional props renders 1`] = `
<div
  className="MOCK FEED TYPE"
>
  <h3>
    Top Albums
  </h3>
  <Order
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="MOCK ORDER"
  />
  <Reset
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="MOCK ORDER"
  />
  <GetLatest
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="MOCK ORDER"
  />
  <GetLatestOrderBy
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="MOCK ORDER"
  />
  <Albums
    items={
      Array [
        Object {
          "artistName": "MOCK ARTIST NAME",
          "artistUrl": "MOCK ARTIST URL",
          "artworkUrl100": "MOCK ARTWORK URL 100",
          "copyright": "MOCK COPYRIGHT",
          "kind": "MOCK KIND",
          "name": "MOCK NAME",
          "releaseDate": "2020-11-16",
          "url": "MOCK URL",
        },
      ]
    }
  />
</div>
`;

exports[`shinkansen-cogs/cogs/checkbox <Component /> With required props renders 1`] = `
<div
  className="MOCK FEED TYPE"
>
  <h3>
    Top Albums
  </h3>
  <Order
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="by-none"
  />
  <Reset
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="by-none"
  />
  <GetLatest
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="by-none"
  />
  <GetLatestOrderBy
    feedType="MOCK FEED TYPE"
    onClick={[Function]}
    order="by-none"
  />
  <Albums
    items={Array []}
  />
</div>
`;
